### ISSUE TITLE
Space filter accessibility refinement

### ISSUE THREAD

**Issue by zebdelk** [cedardevs/onestop#421]
- [x] error message should be specific enough for a screen reader to know which field is wrong and why, for example if N W S are fine but E is out of bounds.
- [x] clear button should have aria-label that specifies 'clear bounding box / location filters'
- [x] apply button should have aria-label that specifies 'apply bounding box / location filters'
- [x] use aria-placeholder, since screen readers are not reporting the input placeholder attribute.
- [x] enter button should apply the filter on each field
- [x] show map button should not be aria hidden (for users who find screen readers easier, but have enough vision to use the draw functionality)
- [x] put expanded true/false attribute on show/hide map button
- [x] move show/hide map button below bounding box text entry section
- [x] icons on the show/hide map button need alt text (to prevent automated tools from reporting meaningless errors)

some additional considerations:
- [x] consider putting the text about how to draw above the map itself, leaving only the description of using the bounding box?
- [x] use H3 instead of H4 for 'Additional Filtering Options' (warning due to H3 level being skipped)

**Comment by zebdelk** [comment]
By the same logic in #420, the information provided for validation errors is probably sufficient already.

**Comment by zebdelk** [comment]
- [x] Also discovered Exclude Global was not presented as an applied filter that could be removed from above the search results, and fixed that.
